proposals/lip-0052.md

```diff
@@ -362,712 +377,1926 @@ The module provides the following commands to modify the NFT store.
-    * For a given `owner` and `tokenID.chainID`, the entries must be in ascending order of `tokenID.collection`.
-    * For a given `owner`, `tokenID.chainID` and `tokenID.collection` the entries must be in ascending order of `tokenID.index`.
-  * For all elements of the `NFTSubstore` array, if the value of `owner` has length 4 (representing an escrowed NFT) then the corresponding `NFTID.chainID` must equal `CHAIN_ID_ALIAS_NATIVE`.
-  * For all elements of the `NFTSubstore` array, if the value of `attributes` must have length less than or equal to `MAX_BYTE_LENGTH_ATTRIBUTES`.
-  * Accross all element of the `collectionSubstore` array, all values given for `collection` must be unique.
-  * The `collectionSubstore` array must be in ascending order of `collection`.
-  * In the `terminatedEscrowSubstore` array, each element must be unique (appears at most once).
-  * The `terminatedEscrowSubstore` array must be in ascending order.
-
-* For each entry `NFTEntry` in `genesisBlockAssetObject.NFTSubstore`, create an entry in the NFT substore with:
-  ```python
-  storeKey = uint32be(NFTEntry.NFTID.chainID) || uint32be(NFTEntry.NFTID.collection) || uint64be(NFTEntry.NFTID.index)
-  storeValue = {"owner": NFTEntry.owner,
-                "lockingModuleID": NFTEntry.lockingModuleID,
-                "attributes": NFTEntry.attributes
-               } serialized using NFTStoreSchema.
-  ```
+    if getChainID(nftID) not in [OWN_CHAIN_ID, sendingChainID]:
+        raise Exception("NFT is not native to either the sending chain or the receiving chain")
 
-  Further, if `NFTEntry.owner` has length 20, create an entry in the user substore with:
+    if getChainID(nftID) == OWN_CHAIN_ID and NFTStore[nftID] entry does not exist:
+        raise Exception("Non-existent entry in the NFT substore")
 
-  ```python
-  storeKey = NFTEntry.owner || uint32be(NFTEntry.NFTID.chainID) || uint32be(NFTEntry.NFTID.collection) || uint64be(NFTEntry.NFTID.index)
-  storeValue = {"exists": True} serialized using userStoreSchema.
-  ```
+    if getChainID(nftID) == OWN_CHAIN_ID and NFTStore[nftID].owner != ccm.sendingChainID:
+        raise Exception("NFT has not been properly escrowed")
 
-* For each entry `collectionEntry` in `genesisBlockAssetObject.collectionSubstore`, create an entry in the collection substore with:
-  ```python
-  storeKey = uint32be(collectionEntry.collection)
-  storeValue = {"nextAvailableIndex": collectionEntry.nextAvailableIndex} serialized using collectionStoreSchema.
-  ```
+    if getChainID(nftID) != OWN_CHAIN_ID and NFTStore[nftID] entry exists:
+        raise Exception("NFT substore entry already exists")
 
-* Create an entry in the available collection substore with:
-  ```python
-  storeKey = EMPTY_BYTES
-  storeValue = {"nextAvailableCollection": genesisBlockAssetObject.availableCollectionSubstore}
-               serialized using availableCollectionStoreSchema.
-  ```
+    if ccm.status > MAX_RESERVED_ERROR_STATUS:
```

Comment:
I would do this check first, so in case of errors we do not need to access the state

Reply:
OK, I will move to the beginning.